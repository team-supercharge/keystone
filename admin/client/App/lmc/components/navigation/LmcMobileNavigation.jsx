import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { isBrowser, isTablet } from 'react-device-detect'
import { GlyphButton } from '../../../elemental'
import LmcMobileNavMenu from './LmcMobileNavMenu.jsx'

export default class LmcMobileNavigation extends Component {
    state = {
        menuIsVisible: false
    }

    toggleMenu = () => {
        this.setState(prevState => ({
            menuIsVisible: !prevState.menuIsVisible
        }))
    }

    renderMenu () {
        const { sections } = this.props
        if (!this.state.menuIsVisible) return null

        return (
            <LmcMobileNavMenu
                sections={sections}
            />
        )
    }

    render () {
        if (isBrowser || isTablet) return null
        const icon = this.state.menuIsVisible ? 'x' : 'three-bars'
        const lmcLogo = `${Keystone.adminPath}/images/lmc-logo-white.svg`

        return (
            <div style={styles.backgroundContainer}>
                <div 
                    className='lmc-theme-gradient' 
                    style={styles.container}
                >
                    <GlyphButton
                        className='lmc-mobile-nav-button'
                        glyph={icon}
                        glyphSize='small'
                        onClick={this.toggleMenu}
                    />
                    <span style={styles.logoContainer}>
                        <img 
                            src={lmcLogo} 
                            width={100}
                            alt="Log my Care" 
                        />
                    </span>
                    { this.renderMenu() }
                </div>
            </div>
        )
    }    
}

const styles = {
    backgroundContainer: {
        height: 40,
        width: '100%',
    },
    button: {
        backgroundColor: 'transparent',
        border: 'none',
        cursor: 'pointer',
    },
    container: {
        height: 40,
        position: 'fixed',
        top: 0,
        width: '100%',
        zIndex: 10,
        textAlign: 'center',
    },
    logoContainer: {
        position: 'relative',
        top: 6,
    }
}

LmcMobileNavigation.propTypes = {
    sections: PropTypes.array.isRequired
}