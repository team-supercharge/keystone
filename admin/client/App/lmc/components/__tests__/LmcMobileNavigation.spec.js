import React from 'react'
import { shallow } from 'enzyme'
import LmcMobileNavigation from '../navigation/LmcMobileNavigation.jsx'

jest.mock('react-device-detect', () => ({
    isBrowser: false,
    isTablet: false
}))

describe('LmcMobileNavigation', () => {
    let wrapper
    let sections

    beforeEach(() => {
        global.Keystone = { adminPath: '/admin' }
        sections = []
        wrapper = shallow(
            <LmcMobileNavigation
                sections={sections}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    
    it('renders an LmcMobileNavMenu when the button is clicked', () => {
        const button = wrapper.find('GlyphButton')
        expect(wrapper.find('LmcMobileNavMenu').length).toEqual(0)

        button.simulate('click')
        expect(wrapper.find('LmcMobileNavMenu').length).toEqual(1)
    })
})