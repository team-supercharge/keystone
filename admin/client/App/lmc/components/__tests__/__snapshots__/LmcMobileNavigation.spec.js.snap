// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcMobileNavigation renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    style={
      Object {
        "height": 40,
        "width": "100%",
      }
    }
  >
    <div
      className="lmc-theme-gradient"
      style={
        Object {
          "height": 40,
          "position": "fixed",
          "textAlign": "center",
          "top": 0,
          "width": "100%",
          "zIndex": 10,
        }
      }
    >
      <GlyphButton
        className="lmc-mobile-nav-button"
        glyph="three-bars"
        glyphSize="small"
        glyphStyle={Object {}}
        onClick={[Function]}
        position="default"
      />
      <span
        style={
          Object {
            "position": "relative",
            "top": 6,
          }
        }
      >
        <img
          alt="Log my Care"
          src="/admin/images/lmc-logo-white.svg"
          width={100}
        />
      </span>
    </div>
  </div>,
  "nodes": Array [
    <div
      style={
        Object {
          "height": 40,
          "width": "100%",
        }
      }
    >
      <div
        className="lmc-theme-gradient"
        style={
          Object {
            "height": 40,
            "position": "fixed",
            "textAlign": "center",
            "top": 0,
            "width": "100%",
            "zIndex": 10,
          }
        }
      >
        <GlyphButton
          className="lmc-mobile-nav-button"
          glyph="three-bars"
          glyphSize="small"
          glyphStyle={Object {}}
          onClick={[Function]}
          position="default"
        />
        <span
          style={
            Object {
              "position": "relative",
              "top": 6,
            }
          }
        >
          <img
            alt="Log my Care"
            src="/admin/images/lmc-logo-white.svg"
            width={100}
          />
        </span>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcMobileNavigation
        sections={Array []}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcMobileNavigation {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "sections": Array [],
        },
        "refs": Object {},
        "state": Object {
          "menuIsVisible": false,
        },
        "toggleMenu": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          style={
            Object {
              "height": 40,
              "width": "100%",
            }
          }
        >
          <div
            className="lmc-theme-gradient"
            style={
              Object {
                "height": 40,
                "position": "fixed",
                "textAlign": "center",
                "top": 0,
                "width": "100%",
                "zIndex": 10,
              }
            }
          >
            <GlyphButton
              className="lmc-mobile-nav-button"
              glyph="three-bars"
              glyphSize="small"
              glyphStyle={Object {}}
              onClick={[Function]}
              position="default"
            />
            <span
              style={
                Object {
                  "position": "relative",
                  "top": 6,
                }
              }
            >
              <img
                alt="Log my Care"
                src="/admin/images/lmc-logo-white.svg"
                width={100}
              />
            </span>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          style={
            Object {
              "height": 40,
              "width": "100%",
            }
          }
        >
          <div
            className="lmc-theme-gradient"
            style={
              Object {
                "height": 40,
                "position": "fixed",
                "textAlign": "center",
                "top": 0,
                "width": "100%",
                "zIndex": 10,
              }
            }
          >
            <GlyphButton
              className="lmc-mobile-nav-button"
              glyph="three-bars"
              glyphSize="small"
              glyphStyle={Object {}}
              onClick={[Function]}
              position="default"
            />
            <span
              style={
                Object {
                  "position": "relative",
                  "top": 6,
                }
              }
            >
              <img
                alt="Log my Care"
                src="/admin/images/lmc-logo-white.svg"
                width={100}
              />
            </span>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcMobileNavigation
    sections={Array []}
  />,
}
`;
